api/v1alpha1/baseboardmanagement_types.go

```diff
@@ -17,29 +17,87 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// PowerState represents power state the BaseboardManagement.
+type PowerState string
+
+// BootDevice represents boot device of the BaseboardManagement.
+type BootDevice string
+
+// BaseboardManagementConditionType represents the condition of the BaseboardManagement.
+type BaseboardManagementConditionType string
```

Comment:
Nit: can you drop `Type`
```suggestion
type BaseboardManagementCondition string
```

Reply:
ah neveremind, I see this is for `BaseboardManagementCondition.Type`. Leave as-is